cloudinit/cmd/devel/logs.py

```diff
@@ -312,25 +302,37 @@ def collect_logs(
         init.read_cfg()
         paths = get_log_paths(init)
 
-        _collect_version_info(log_dir, verbosity)
-        _collect_system_logs(log_dir, verbosity)
-        _collect_cloudinit_logs(
-            log_dir, verbosity, init, paths, include_userdata
-        )
-        _collect_installer_logs(log_dir, include_userdata, verbosity)
-        _collect_run_dir(log_dir, verbosity, paths)
+        _collect_version_info(log_dir)
+        _collect_system_logs(log_dir)
+        _collect_cloudinit_logs(log_dir, init, paths, include_userdata)
+        _collect_installer_logs(log_dir, include_userdata)
+        _collect_run_dir(log_dir, paths)
         with chdir(tmp_dir):
             subp(["tar", "czvf", tarfile, log_dir.replace(f"{tmp_dir}/", "")])
-    sys.stderr.write("Wrote %s\n" % tarfile)
+    LOG.info("Wrote %s", tarfile)
     return 0
 
 
+def _setup_logger(verbosity: int) -> None:
+    log.reset_logging()
+    if verbosity == 0:
+        level = logging.INFO
+    elif verbosity == 1:
+        level = logging.DEBUG
+    else:
+        level = log.TRACE
+    LOG.setLevel(level)
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter("%(message)s"))
```

Comment:
I prefer this a lot over `sys.stderr.write()` all over the place. Are you thinking about replacing manual stderr writes with a StreamHandler in other parts of cloud-init?

Reply:
I wasn't thinking of that. I honestly don't know when or why we log to stderr in other places so I'd have to grok that first to know if something like this is a suitable alternative.

Reply:
> I wasn't thinking of that, but it sounds like a good idea. I honestly don't know when or why we log to stderr in other places so I'd have to grok that first to know if something like this is a suitable alternative.


iirc the places that I've seen them besides in under `cmd/` usually had comments about it existing as a fallback to ensure errors get to a serial console in case the log file fails to get written - the reasonable thing to do in error paths - just not using the best interfaces imo